pdns/lwres.cc

```diff
@@ -343,70 +346,91 @@ LWResult::Result asyncresolve(const ComboAddress& ip, const DNSName& domain, int
   }
   else {
     try {
-      const struct timeval timeout{ g_networkTimeoutMsec / 1000, static_cast<suseconds_t>(g_networkTimeoutMsec) % 1000 * 1000};
-
-      Socket s(ip.sin4.sin_family, SOCK_STREAM);
-      s.setNonBlocking();
-      localip = pdns::getQueryLocalAddress(ip.sin4.sin_family, 0);
-      s.bind(localip);
-
-      std::shared_ptr<TLSCtx> tlsCtx{nullptr};
-      if (SyncRes::s_dot_to_port_853 && ip.getPort() == 853) {
-        TLSContextParameters tlsParams;
-        tlsParams.d_provider = "openssl";
-        tlsParams.d_validateCertificates = false;
-        //tlsParams.d_caStore = caaStore;
-        tlsCtx = getTLSContext(tlsParams);
-        if (tlsCtx == nullptr) {
-          g_log << Logger::Error << "DoT to " << ip << " requested but not available" << endl;
+      while (true) {
```

Comment:
I wonder if we could turn the body of that loop into a function, which would not care about whether the connection is new and the loop would handle the retry. Of course that means creating a new function handling the part common to the UDP and TCP path but it might not be a bad thing anyway?

Reply:
Latest commit does refactoring, but not as far as you suggest. That is for a next step I'd say.